impl/src/main/java/org/jboss/weld/bean/DisposalMethod.java

```diff
@@ -123,7 +123,7 @@ private void checkDisposalMethod(EnhancedAnnotatedMethod<T, ? super X> enhancedA
         }
         if (declaringBean instanceof SessionBean<?>) {
             SessionBean<?> sessionBean = (SessionBean<?>) declaringBean;
-            Set<MethodSignature> businessMethodSignatures = sessionBean.getBusinessMethodSignatures();
+            Set<MethodSignature> businessMethodSignatures = sessionBean.getLocalBusinessMethodSignatures();
```

Comment:
I wonder whether a remote business method may also be a producer/disposer method?